ath79: add TP-Link TL-WR710N v1

marcin1j opened the pull request:
This commit adds support for TP-Link TL-WR710N v1 router.

CPU: Atheros AR9331 400MHz
RAM: 32MB
FLASH: 8MiB
PORTS: 1 Port 100/10 LAN (connected to a switch), 1 Port 100/10 WAN
WiFi: Atheros AR9331 1x2:1 bgn
USB: ChipIdea HDRC USB2.0
LED: SYS
BTN: Reset

Sysupgrade from `ar71xx` works without glitches. Only wireless radio path needs adjustment afterwards.
Network interfaces assigned for LAN and WAN ports are `eth1` and `eth0` respectively, what's consistent with `ar71xx` target.

marcin1j committed:
ath79: add TP-Link TL-WR710N v1

This commit adds support for TP-Link TL-WR710N v1 router.

CPU: Atheros AR9331 400MHz
RAM: 32MB
FLASH: 8MiB
PORTS: 1 Port 100/10 LAN (connected to a switch), 1 Port 100/10 WAN
WiFi: Atheros AR9331 1x2:1 bgn
USB: ChipIdea HDRC USB2.0
LED: SYS
BTN: Reset

Sysupgrade from `ar71xx` works without glitches.
Network interfaces assigned for LAN and WAN ports are `eth1` and `eth0`
respectively, what's consistent with `ar71xx` target. Wireless radio
path is automatically upgraded from `platform/ar933x_wmac` to
`platform/ahb/18100000.wmac`.

Signed-off-by: Marcin Jurkowski <marcin1j@gmail.com>

chunkeey commented on target/linux/ath79/base-files/etc/board.d/02_network:
```diff
@@ -181,6 +181,9 @@ ath79_setup_interfaces()
 	tplink,tl-wr941-v2)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4" "wan"
 		;;
+	tplink,tl-wr710n-v1)
```
The ubnt,routerstation and the glinet,ar150, glinet,ar300m and glinet,gl-x750 seem to share the same setting.
I wonder why nobody has tried to merge them together.

marcin1j replied:
Now it's merged with glinet etc

chunkeey commented on target/linux/ath79/dts/ar9331_tplink_tl-wr710n-v1.dts:
```diff
@@ -0,0 +1,130 @@
+		led-upgrade = &led_system;
+	};
+
+	gpio-keys-polled {
```
"keys" is enough. (And recommended by the [DeviceTree spec](https://github.com/devicetree-org/devicetree-specification/blob/master/source/devicetree-basics.rst#generic-names-recommendation)])

marcin1j replied:
Changed

chunkeey commented on target/linux/ath79/dts/ar9331_tplink_tl-wr710n-v1.dts:
```diff
@@ -0,0 +1,130 @@
+		};
+	};
+
+	gpio-leds {
```
"leds". (And recommended by the DeviceTree spec])

marcin1j replied:
Changed

chunkeey commented on target/linux/ath79/dts/ar9331_tplink_tl-wr710n-v1.dts:
```diff
@@ -0,0 +1,130 @@
+		};
+	};
+
+	reg_usb_vbus: reg_usb_vbus {
```
maybe reg_usb_vbus: regulator { ?

marcin1j replied:
Changed

mpratt14 replied:
@chunkeey @marcin1j  USB power is broken for me. I don't know if the problem is DTS or the GPIO

mpratt14 replied:
Nevermind, for some reason in my build config there is no kmod-usb-core or kmod-usb-chipidea2 .....

chunkeey reviewed (commented):
This looks fine. Though the PR is a bit overshadowed by the recent mailing-list discussion about stopping to add [new 4MiB Flash OR 32 MiB RAM devices](http://lists.infradead.org/pipermail/openwrt-devel/2018-December/014926.html).

CodeFetch commented:
@chunkeey This is not a "new" device support. As far as I understood devices with existing support in ar71xx should still be ported to DTS. There are many projects that use modified OpenWrt versions without Luci etc. that are not affected by these size constraints. E.g. Freifunk still uses about 20,000 4MB/32MB devices and they should run until they break.

ynezz commented:
I think, that this PR falls in `DTS based targets` case ([quote of @pepe2k from](http://lists.infradead.org/pipermail/openwrt-devel/2018-December/014942.html)):

> In case of DTS based targets (where single device support in code 
doesn't have that big impact like in, for example, ar71xx) and good 
quality patch/PR I don't see a reason to not merge the support.

And [quote from @jow-](http://lists.infradead.org/pipermail/openwrt-devel/2018-December/015152.html)

> I'm against rejecting such submissions if they're self-contained enough
(e.g. for DTS based targets as mentioned by @pepe2k).

> I'd favor accepting the basic board support for such boards but
disabling the image generation (commented out).

chunkeey commented on target/linux/ath79/image/generic-tp-link.mk:
```diff
@@ -99,6 +99,16 @@ define Device/tplink_tl-wr1043nd-v1
+  $(Device/tplink-8mlzma)
+  ATH_SOC := ar9331
+  DEVICE_TITLE := TP-Link TL-WR710N v1
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb-chipidea2 kmod-usb-ledtrig-usbport
```
> swconfig is needed for VLANs on LAN interface.

Ok, this smells a bit broken. Nowadays, it should be possible to use the kernel's generic vlan
tagging/untagging support to do that. 

Try:
```
     # ip link add link eth0 name eth0.1 type vlan id 1 
     # ip link show
```

This will create a new  virtual lan ethernet device for vlanid 1 on eth0.

ynezz reviewed (changes requested):
> Only wireless radio path needs adjustment afterwards.

@marcin1j Do you think, that you can come up with some solution for this? I've already suggested something similar in this mail thread http://lists.infradead.org/pipermail/openwrt-devel/2019-March/016166.html

If you're not going to handle this yourself (you're not obliged to do this, but it would be nice), could you please at least provide some details about this upgrade issue, so it could be fixed?

Please can you as well rebase this PR, build it, test it and provide here the boot log for the reference? I'm planing to merge this. Thanks.

marcin1j commented:
@ynezz 
I'll rebase this PR against master and test it again. As for wireless config upgrade issue: if it's already agreed how to handle conversion (and from mailing list it's clear there is a "prior art") than I can provide reasonable  `target/linux/ath79/base-files/etc/uci-defaults/03_wireless` that will do the job.

tmn505 commented:
Hi Marcin,
@ynezz directed me on IRC to this PR. Regarding conversion of wireless path, there is recently merged migration event for this `target/linux/ath79/base-files/etc/hotplug.d/ieee80211/00-wmac-migration`, so it should be easy to just add Your device case.

marcin1j commented:
@tmn505 Thanks for the hint. Now this is really trivial.
@ynezz I'll test everything (including WMAC migration) by Sunday.

marcin1j commented:
@ynezz Everything is rebased and tested. No problems. Automatic wireless path migration also works.

Boot log:
```
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.000000] Linux version 4.14.106 (marcin@tower) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r9059+1-6e104c63d6)) #0 Fri Mar 22 23:59:52 2019
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] bootconsole [early0] enabled
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] CPU0 revision is: 00019374 (MIPS 24Kc)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] MIPS: machine is TP-Link TL-WR710N v1
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] SoC: Atheros AR9330 rev 1
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Determined physical RAM map:
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000]  memory: 02000000 @ 00000000 (usable)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Initrd not found or empty - disabling initrd
Sat Mar 23 00:59:57 2019 kern.warn kernel: [    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
Sat Mar 23 00:59:57 2019 kern.warn kernel: [    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Zone ranges:
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000]   Normal   [mem 0x0000000000000000-0x0000000001ffffff]
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Movable zone start for each node
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Early memory node ranges
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000]   node   0: [mem 0x0000000000000000-0x0000000001ffffff]
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000001ffffff]
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.000000] On node 0 totalpages: 8192
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.000000] free_area_init_node: node 0, pgdat 80527db0, node_mem_map 81005fe0
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.000000]   Normal zone: 64 pages used for memmap
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.000000]   Normal zone: 0 pages reserved
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.000000]   Normal zone: 8192 pages, LIFO batch:0
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.000000] random: get_random_bytes called from start_kernel+0x8c/0x47c with crng_init=0
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.000000] pcpu-alloc: [0] 0
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 8128
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.000000] Kernel command line: console=ttyATH0,115200 rootfstype=squashfs,jffs2
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] PID hash table entries: 128 (order: -3, 512 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Dentry cache hash table entries: 4096 (order: 2, 16384 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Inode-cache hash table entries: 2048 (order: 1, 8192 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Writing ErrCtl register=00000000
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Readback ErrCtl register=00000000
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] Memory: 25664K/32768K available (3830K kernel code, 150K rwdata, 920K rodata, 1236K init, 204K bss, 7104K reserved, 0K cma-reserved)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] NR_IRQS: 51
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] CPU clock: 400.000 MHz
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 9556302233 ns
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.000014] sched_clock: 32 bits at 200MHz, resolution 5ns, wraps every 10737418237ns
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.007568] Calibrating delay loop... 265.42 BogoMIPS (lpj=1327104)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.092165] pid_max: default: 32768 minimum: 301
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.096882] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.102935] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.115311] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.123414] futex hash table entries: 256 (order: -1, 3072 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.129298] pinctrl core: initialized pinctrl subsystem
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.135575] NET: Registered protocol family 16
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.173891] clocksource: Switched to clocksource MIPS
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.178909] NET: Registered protocol family 2
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.182815] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.188434] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.194486] TCP: Hash tables configured (established 1024 bind 1024)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.200726] UDP hash table entries: 256 (order: 0, 4096 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.206167] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.212426] NET: Registered protocol family 1
Sat Mar 23 00:59:57 2019 kern.debug kernel: [    0.216339] PCI: CLS 0 bytes, default 32
Sat Mar 23 00:59:57 2019 kern.warn kernel: [    0.221344] Crashlog allocated RAM at address 0x1f00000
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.226821] workingset: timestamp_bits=30 max_order=13 bucket_order=0
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.238946] squashfs: version 4.0 (2009/01/31) Phillip Lougher
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.243140] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.264863] io scheduler noop registered
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.267228] io scheduler deadline registered (default)
Sat Mar 23 00:59:57 2019 kern.err kernel: [    0.272556] ar7200-usb-phy usb-phy: phy reset is missing
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.277920] pinctrl-single 18040028.pinmux: 64 pins at pa b8040028 size 8
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.285194] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.291002] 18020000.uart: ttyATH0 at MMIO 0x18020000 (irq = 8, base_baud = 1562500) is a AR933X UART
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.299234] console [ttyATH0] enabled
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.305998] bootconsole [early0] disabled
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.333544] m25p80 spi0.0: s25fl064k (8192 Kbytes)
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.337001] 3 fixed-partitions partitions found on MTD device spi0.0
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.343221] Creating 3 MTD partitions on "spi0.0":
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.348033] 0x000000000000-0x000000020000 : "u-boot"
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.354305] 0x000000020000-0x0000007f0000 : "firmware"
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.359306] 2 tplink-fw partitions found on MTD device firmware
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.364050] Creating 2 MTD partitions on "firmware":
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.368933] 0x000000000000-0x00000018f562 : "kernel"
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.374956] 0x00000018f564-0x0000007d0000 : "rootfs"
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.379817] mtd: device 3 (rootfs) set to be root filesystem
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.384575] 1 squashfs-split partitions found on MTD device rootfs
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.390635] 0x0000005c0000-0x0000007d0000 : "rootfs_data"
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    0.397081] 0x0000007f0000-0x000000800000 : "art"
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.403294] libphy: Fixed MDIO Bus: probed
Sat Mar 23 00:59:57 2019 kern.info kernel: [    0.744374] libphy: ag71xx_mdio: probed
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.033962] mdio-bus.0:1f: Found an AR7240/AR9330 built-in switch
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.038846] libphy: ar7240sw_mdio: probed
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.425572] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd041, driver=Generic PHY]
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.434613] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.775906] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.784415] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.792013] NET: Registered protocol family 10
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.800633] Segment Routing with IPv6
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.802973] NET: Registered protocol family 17
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.807442] 8021q: 802.1Q VLAN Support v1.8
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.814664] usb_vbus: disabling
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.821322] VFS: Mounted root (squashfs filesystem) readonly on device 31:3.
Sat Mar 23 00:59:57 2019 kern.info kernel: [    1.836434] Freeing unused kernel memory: 1236K
Sat Mar 23 00:59:57 2019 kern.warn kernel: [    1.839502] This architecture does not have kernel memory protection.
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    2.543907] random: fast init done
Sat Mar 23 00:59:57 2019 user.info kernel: [    2.636991] init: Console is alive
Sat Mar 23 00:59:57 2019 user.info kernel: [    4.393362] kmodloader: loading kernel modules from /etc/modules-boot.d/*
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.553062] usbcore: registered new interface driver usbfs
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.557305] usbcore: registered new interface driver hub
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.562526] usbcore: registered new device driver usb
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.578474] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.592123] ci_hdrc ci_hdrc.0: EHCI Host Controller
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.595725] ci_hdrc ci_hdrc.0: new USB bus registered, assigned bus number 1
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.623917] ci_hdrc ci_hdrc.0: USB 2.0 started, EHCI 1.00
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.629070] hub 1-0:1.0: USB hub found
Sat Mar 23 00:59:57 2019 kern.info kernel: [    4.632059] hub 1-0:1.0: 1 port detected
Sat Mar 23 00:59:57 2019 user.info kernel: [    4.637931] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
Sat Mar 23 00:59:57 2019 user.info kernel: [    4.646194] init: - preinit -
Sat Mar 23 00:59:57 2019 kern.info kernel: [    5.527713] usb 1-1: new full-speed USB device number 2 using ci_hdrc
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    5.795182] random: procd: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    6.130398] random: jshn: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    6.417196] random: jshn: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 kern.warn kernel: [    6.861387] urandom_read: 2 callbacks suppressed
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    6.861401] random: jshn: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    6.948055] random: jshn: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 kern.notice kernel: [    7.045672] random: jshn: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    7.689728] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
Sat Mar 23 00:59:57 2019 kern.info kernel: [    7.694332] IPv6: ADDRCONF(NETDEV_UP): eth1.1: link is not ready
Sat Mar 23 00:59:57 2019 kern.info kernel: [    8.735071] eth1: link up (1000Mbps/Full duplex)
Sat Mar 23 00:59:57 2019 kern.info kernel: [    8.738274] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
Sat Mar 23 00:59:57 2019 kern.info kernel: [    8.745177] IPv6: ADDRCONF(NETDEV_CHANGE): eth1.1: link becomes ready
Sat Mar 23 00:59:57 2019 user.info kernel: [   10.886584] mount_root: loading kmods from internal overlay
Sat Mar 23 00:59:57 2019 user.info kernel: [   10.936720] kmodloader: loading kernel modules from //etc/modules-boot.d/*
Sat Mar 23 00:59:57 2019 user.info kernel: [   10.943042] kmodloader: done loading kernel modules from //etc/modules-boot.d/*
Sat Mar 23 00:59:57 2019 kern.warn kernel: [   11.202903] jffs2_scan_eraseblock(): End of filesystem marker found at 0x20000
Sat Mar 23 00:59:57 2019 kern.warn kernel: [   11.208735] jffs2_build_filesystem(): unlocking the mtd device...
Sat Mar 23 00:59:57 2019 kern.warn kernel: [   11.208808] done.
Sat Mar 23 00:59:57 2019 kern.warn kernel: [   11.216745] jffs2_build_filesystem(): erasing all blocks after the end marker...
Sat Mar 23 00:59:57 2019 kern.warn kernel: [   16.867524] done.
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   16.875574] jffs2: notice: (453) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
Sat Mar 23 00:59:57 2019 user.info kernel: [   16.891490] block: attempting to load /tmp/jffs_cfg/upper/etc/config/fstab
Sat Mar 23 00:59:57 2019 user.err kernel: [   16.898065] block: unable to load configuration (fstab: Entry not found)
Sat Mar 23 00:59:57 2019 user.info kernel: [   16.904718] block: attempting to load /tmp/jffs_cfg/etc/config/fstab
Sat Mar 23 00:59:57 2019 user.err kernel: [   16.910928] block: unable to load configuration (fstab: Entry not found)
Sat Mar 23 00:59:57 2019 user.info kernel: [   16.917628] block: attempting to load /etc/config/fstab
Sat Mar 23 00:59:57 2019 user.err kernel: [   16.922885] block: unable to load configuration (fstab: Entry not found)
Sat Mar 23 00:59:57 2019 user.err kernel: [   16.929494] block: no usable configuration
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   16.976357] jffs2: notice: (451) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
Sat Mar 23 00:59:57 2019 user.info kernel: [   17.138267] block: attempting to load /tmp/jffs_cfg/upper/etc/config/fstab
Sat Mar 23 00:59:57 2019 user.err kernel: [   17.144091] block: unable to load configuration (fstab: Entry not found)
Sat Mar 23 00:59:57 2019 user.info kernel: [   17.150551] block: attempting to load /tmp/jffs_cfg/etc/config/fstab
Sat Mar 23 00:59:57 2019 user.err kernel: [   17.157087] block: unable to load configuration (fstab: Entry not found)
Sat Mar 23 00:59:57 2019 user.info kernel: [   17.163509] block: attempting to load /etc/config/fstab
Sat Mar 23 00:59:57 2019 user.err kernel: [   17.168949] block: unable to load configuration (fstab: Entry not found)
Sat Mar 23 00:59:57 2019 user.err kernel: [   17.175588] block: no usable configuration
Sat Mar 23 00:59:57 2019 user.info kernel: [   17.181411] mount_root: overlay filesystem has not been fully initialized yet
Sat Mar 23 00:59:57 2019 user.info kernel: [   17.189360] mount_root: switching to jffs2 overlay
Sat Mar 23 00:59:57 2019 kern.warn kernel: [   17.234759] overlayfs: upper fs does not support tmpfile.
Sat Mar 23 00:59:57 2019 user.warn kernel: [   18.811138] urandom-seed: Seed file not found (/etc/urandom.seed)
Sat Mar 23 00:59:57 2019 kern.info kernel: [   18.973097] eth1: link down
Sat Mar 23 00:59:57 2019 user.info kernel: [   19.001799] procd: - early -
Sat Mar 23 00:59:57 2019 user.info kernel: [   19.751601] procd: - ubus -
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   20.029695] random: ubusd: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   20.035539] random: ubusd: uninitialized urandom read (4 bytes read)
Sat Mar 23 00:59:57 2019 user.info kernel: [   20.043012] procd: - init -
Sat Mar 23 00:59:57 2019 user.info kernel: [   20.632474] kmodloader: loading kernel modules from /etc/modules.d/*
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   20.640019] Key type dns_resolver registered
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.665237] RPC: Registered named UNIX socket transport module.
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.669704] RPC: Registered udp transport module.
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.674470] RPC: Registered tcp transport module.
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.679078] RPC: Registered tcp NFSv4.1 backchannel transport module.
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   20.770163] NFS: Registering the id_resolver key type
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   20.773804] Key type id_resolver registered
Sat Mar 23 00:59:57 2019 kern.notice kernel: [   20.778022] Key type id_legacy registered
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.793549] ip6_tables: (C) 2000-2006 Netfilter Core Team
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.814075] Loading modules backported from Linux version v4.19.23-0-g67d52fae61c1
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.820199] Backport generated by backports.git v4.19.23-1-0-g480a925a
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.829883] ip_tables: (C) 2000-2006 Netfilter Core Team
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.849257] nf_conntrack version 0.5.0 (1024 buckets, 4096 max)
Sat Mar 23 00:59:57 2019 kern.info kernel: [   20.956209] xt_time: kernel timezone is -0000
Sat Mar 23 00:59:57 2019 kern.info kernel: [   21.148612] usbcore: registered new interface driver snd-usb-audio
Sat Mar 23 00:59:57 2019 kern.debug kernel: [   21.229389] ath: EEPROM regdomain: 0x0
Sat Mar 23 00:59:57 2019 kern.debug kernel: [   21.229404] ath: EEPROM indicates default country code should be used
Sat Mar 23 00:59:57 2019 kern.debug kernel: [   21.229410] ath: doing EEPROM country->regdmn map search
Sat Mar 23 00:59:57 2019 kern.debug kernel: [   21.229429] ath: country maps to regdmn code: 0x3a
Sat Mar 23 00:59:57 2019 kern.debug kernel: [   21.229437] ath: Country alpha2 being used: US
Sat Mar 23 00:59:57 2019 kern.debug kernel: [   21.229443] ath: Regpair used: 0x3a
Sat Mar 23 00:59:57 2019 kern.debug kernel: [   21.244514] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
Sat Mar 23 00:59:57 2019 kern.info kernel: [   21.247199] ieee80211 phy0: Atheros AR9330 Rev:1 mem=0xb8100000, irq=2
Sat Mar 23 00:59:57 2019 user.info kernel: [   21.304562] kmodloader: done loading kernel modules from /etc/modules.d/*
Sat Mar 23 00:59:58 2019 authpriv.info dropbear[881]: Not backgrounding
Sat Mar 23 01:00:01 2019 user.notice : Added device handler type: 8021ad
Sat Mar 23 01:00:01 2019 user.notice : Added device handler type: 8021q
Sat Mar 23 01:00:01 2019 user.notice : Added device handler type: macvlan
Sat Mar 23 01:00:01 2019 user.notice : Added device handler type: veth
Sat Mar 23 01:00:01 2019 user.notice : Added device handler type: bridge
Sat Mar 23 01:00:01 2019 user.notice : Added device handler type: Network device
Sat Mar 23 01:00:01 2019 user.notice : Added device handler type: tunnel
Sat Mar 23 01:00:03 2019 daemon.err block: /dev/mtdblock3 is already mounted on /rom
Sat Mar 23 01:00:03 2019 daemon.err block: /dev/mtdblock4 is already mounted on /overlay
Sat Mar 23 01:00:06 2019 user.notice ucitrack: Setting up /etc/config/network reload dependency on /etc/config/dhcp
Sat Mar 23 01:00:06 2019 user.notice ucitrack: Setting up /etc/config/network reload dependency on /etc/config/radvd
Sat Mar 23 01:00:06 2019 user.notice ucitrack: Setting up /etc/config/wireless reload dependency on /etc/config/network
Sat Mar 23 01:00:07 2019 user.notice ucitrack: Setting up /etc/config/firewall reload dependency on /etc/config/luci-splash
Sat Mar 23 01:00:07 2019 user.notice ucitrack: Setting up /etc/config/firewall reload dependency on /etc/config/qos
Sat Mar 23 01:00:07 2019 user.notice ucitrack: Setting up /etc/config/firewall reload dependency on /etc/config/miniupnpd
Sat Mar 23 01:00:07 2019 user.notice ucitrack: Setting up /etc/config/dhcp reload dependency on /etc/config/odhcpd
Sat Mar 23 01:00:08 2019 user.notice ucitrack: Setting up /etc/config/fstab reload trigger for non-procd /etc/init.d/fstab
Sat Mar 23 01:00:08 2019 user.notice ucitrack: Setting up /etc/config/system reload trigger for non-procd /etc/init.d/led
Sat Mar 23 01:00:08 2019 user.notice ucitrack: Setting up /etc/config/system reload dependency on /etc/config/luci_statistics
Sat Mar 23 01:00:09 2019 user.notice ucitrack: Setting up /etc/config/system reload dependency on /etc/config/dhcp
Sat Mar 23 01:00:09 2019 kern.info kernel: [   38.372451] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
Sat Mar 23 01:00:09 2019 kern.info kernel: [   38.424831] br-iot: port 1(eth0.7) entered blocking state
Sat Mar 23 01:00:09 2019 kern.info kernel: [   38.428785] br-iot: port 1(eth0.7) entered disabled state
Sat Mar 23 01:00:09 2019 kern.info kernel: [   38.434781] device eth0.7 entered promiscuous mode
Sat Mar 23 01:00:09 2019 kern.info kernel: [   38.438935] device eth0 entered promiscuous mode
Sat Mar 23 01:00:10 2019 daemon.notice netifd: Interface 'iot' is enabled
Sat Mar 23 01:00:10 2019 kern.info kernel: [   38.504256] IPv6: ADDRCONF(NETDEV_UP): br-iot: link is not ready
Sat Mar 23 01:00:10 2019 daemon.notice netifd: Interface 'iot' is setting up now
Sat Mar 23 01:00:10 2019 daemon.notice netifd: Interface 'iot' is now up
Sat Mar 23 01:00:10 2019 kern.info kernel: [   39.391463] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
Sat Mar 23 01:00:10 2019 kern.info kernel: [   39.397143] eth1: link up (1000Mbps/Full duplex)
Sat Mar 23 01:00:10 2019 kern.info kernel: [   39.400472] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
Sat Mar 23 01:00:10 2019 kern.info kernel: [   39.440139] br-lan: port 1(eth1.1) entered blocking state
Sat Mar 23 01:00:10 2019 kern.info kernel: [   39.444177] br-lan: port 1(eth1.1) entered disabled state
Sat Mar 23 01:00:10 2019 kern.info kernel: [   39.450222] device eth1.1 entered promiscuous mode
Sat Mar 23 01:00:10 2019 kern.info kernel: [   39.454337] device eth1 entered promiscuous mode
Sat Mar 23 01:00:11 2019 kern.info kernel: [   39.490021] br-lan: port 1(eth1.1) entered blocking state
Sat Mar 23 01:00:11 2019 kern.info kernel: [   39.494050] br-lan: port 1(eth1.1) entered forwarding state
Sat Mar 23 01:00:11 2019 daemon.notice netifd: Interface 'lan' is enabled
Sat Mar 23 01:00:11 2019 daemon.notice netifd: Interface 'lan' is setting up now
Sat Mar 23 01:00:11 2019 daemon.notice netifd: Interface 'lan' is now up
Sat Mar 23 01:00:11 2019 daemon.info logread[835]: Logread connected to X.X.X.X:514
Sat Mar 23 01:00:12 2019 kern.info kernel: [   40.484529] br-lan: port 2(eth0.1) entered blocking state
Sat Mar 23 01:00:12 2019 kern.info kernel: [   40.488480] br-lan: port 2(eth0.1) entered disabled state
Sat Mar 23 01:00:12 2019 kern.info kernel: [   40.522844] device eth0.1 entered promiscuous mode
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Interface 'loopback' is enabled
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Interface 'loopback' is setting up now
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Interface 'loopback' is now up
Sat Mar 23 01:00:12 2019 kern.info kernel: [   40.566682] eth0: link up (100Mbps/Full duplex)
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Network device 'eth0' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: bridge 'br-lan' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Interface 'lan' has link connectivity
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Network device 'eth1' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: VLAN 'eth1.1' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Network device 'lo' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Interface 'loopback' has link connectivity
Sat Mar 23 01:00:12 2019 kern.info kernel: [   41.157819] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
Sat Mar 23 01:00:12 2019 kern.info kernel: [   41.195227] br-lan: port 2(eth0.1) entered blocking state
Sat Mar 23 01:00:12 2019 kern.info kernel: [   41.199181] br-lan: port 2(eth0.1) entered forwarding state
Sat Mar 23 01:00:12 2019 kern.info kernel: [   41.205408] br-iot: port 1(eth0.7) entered blocking state
Sat Mar 23 01:00:12 2019 kern.info kernel: [   41.210113] br-iot: port 1(eth0.7) entered forwarding state
Sat Mar 23 01:00:12 2019 daemon.notice netifd: VLAN 'eth0.1' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: VLAN 'eth0.7' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: bridge 'br-iot' link is up
Sat Mar 23 01:00:12 2019 daemon.notice netifd: Interface 'iot' has link connectivity
Sat Mar 23 01:00:12 2019 kern.info kernel: [   41.444052] IPv6: ADDRCONF(NETDEV_CHANGE): br-iot: link becomes ready
Sat Mar 23 01:00:18 2019 user.notice firewall: Reloading firewall due to ifup of lan (br-lan)
Sat Mar 23 01:00:24 2019 kern.debug kernel: [   53.208054] ath: EEPROM regdomain: 0x8268
Sat Mar 23 01:00:24 2019 kern.debug kernel: [   53.208069] ath: EEPROM indicates we should expect a country code
Sat Mar 23 01:00:24 2019 kern.debug kernel: [   53.208080] ath: doing EEPROM country->regdmn map search
Sat Mar 23 01:00:24 2019 kern.debug kernel: [   53.208089] ath: country maps to regdmn code: 0x37
Sat Mar 23 01:00:24 2019 kern.debug kernel: [   53.208097] ath: Country alpha2 being used: PL
Sat Mar 23 01:00:24 2019 kern.debug kernel: [   53.208102] ath: Regpair used: 0x37
Sat Mar 23 01:00:24 2019 kern.debug kernel: [   53.208112] ath: regdomain 0x8268 dynamically updated by user
Sat Mar 23 01:00:27 2019 daemon.info procd: - init complete -
Sat Mar 23 01:00:28 2019 daemon.err hostapd: Configuration file: /var/run/hostapd-phy0.conf
Sat Mar 23 01:00:28 2019 kern.info kernel: [   56.966918] IPv6: ADDRCONF(NETDEV_UP): wlan0: link is not ready
Sat Mar 23 01:00:28 2019 kern.info kernel: [   56.979499] br-lan: port 3(wlan0) entered blocking state
Sat Mar 23 01:00:28 2019 kern.info kernel: [   56.983367] br-lan: port 3(wlan0) entered disabled state
Sat Mar 23 01:00:28 2019 kern.info kernel: [   56.989265] device wlan0 entered promiscuous mode
Sat Mar 23 01:00:28 2019 daemon.notice hostapd: wlan0: interface state UNINITIALIZED->COUNTRY_UPDATE
Sat Mar 23 01:00:28 2019 daemon.notice hostapd: wlan0: interface state COUNTRY_UPDATE->HT_SCAN
Sat Mar 23 01:31:38 2019 daemon.err hostapd: Using interface wlan0 with hwaddr 10:XX:XX:XX:XX:XX and ssid "XXX"
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.086460] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.091632] br-lan: port 3(wlan0) entered blocking state
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.096758] br-lan: port 3(wlan0) entered forwarding state
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.225725] br-iot: port 2(wlan0-1) entered blocking state
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.229765] br-iot: port 2(wlan0-1) entered disabled state
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.235841] device wlan0-1 entered promiscuous mode
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.264485] IPv6: ADDRCONF(NETDEV_UP): wlan0-1: link is not ready
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.269195] br-iot: port 2(wlan0-1) entered blocking state
Sat Mar 23 01:31:39 2019 kern.info kernel: [   59.274676] br-iot: port 2(wlan0-1) entered forwarding state
Sat Mar 23 01:31:39 2019 daemon.err hostapd: Using interface wlan0-1 with hwaddr 12:XX:XX:XX:XX:XX and ssid "YYY"
Sat Mar 23 01:31:40 2019 kern.info kernel: [   60.068651] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0-1: link becomes ready
Sat Mar 23 01:31:40 2019 daemon.notice hostapd: wlan0: interface state HT_SCAN->ENABLED
Sat Mar 23 01:31:40 2019 daemon.notice hostapd: wlan0: AP-ENABLED
Sat Mar 23 01:31:40 2019 daemon.info hostapd: wlan0-1: STA 99:99:99:99:99:99 IEEE 802.11: authenticated
Sat Mar 23 01:31:40 2019 daemon.info hostapd: wlan0-1: STA 99:99:99:99:99:99 IEEE 802.11: associated (aid 1)
Sat Mar 23 01:31:40 2019 daemon.notice hostapd: wlan0-1: AP-STA-CONNECTED 99:99:99:99:99:99
Sat Mar 23 01:31:40 2019 daemon.info hostapd: wlan0-1: STA 99:99:99:99:99:99 WPA: pairwise key handshake completed (RSN)
Sat Mar 23 01:31:40 2019 daemon.notice netifd: Network device 'wlan0' link is up
Sat Mar 23 01:31:40 2019 daemon.notice netifd: Network device 'wlan0-1' link is up
Sat Mar 23 01:31:42 2019 daemon.info hostapd: wlan0-1: STA 99:99:99:99:99:99 IEEE 802.11: authenticated
Sat Mar 23 01:31:42 2019 daemon.info hostapd: wlan0-1: STA 99:99:99:99:99:99 IEEE 802.11: associated (aid 2)
Sat Mar 23 01:31:42 2019 daemon.notice hostapd: wlan0-1: AP-STA-CONNECTED 99:99:99:99:99:99
Sat Mar 23 01:31:42 2019 daemon.info hostapd: wlan0-1: STA 99:99:99:99:99:99 WPA: pairwise key handshake completed (RSN)
Sat Mar 23 01:32:12 2019 kern.notice kernel: [   91.836658] random: crng init done
Sat Mar 23 01:32:12 2019 daemon.info urandom_seed[1538]: Seed saved (/etc/urandom.seed)
```

ynezz commented:
Thanks! I've merged it to my staging tree https://git.openwrt.org/?p=openwrt/staging/ynezz.git;a=shortlog;h=refs/heads/staging

adrianschmutzler commented:
Although this PR has been has been closed already, I would be interested in some additional data about the device(s) in case you still have access to it:

I'm currently working on a PR to store the label-mac-address in the DTS (https://github.com/openwrt/openwrt/pull/2159).

I would appreciate if you could provide information about your device:
Please run
`find /proc/device-tree/ -name "*mac-address*"`
on your device (just with the current firmware, no need to update anything; just has to be ath79, no ar71xx).
For each of the returned paths (if there are any), retrieve the mac-address with `cat`, e.g.
`cat /proc/device-tree/ahb/eth@19000000/mac-address`

Now check which of those returned MAC addresses matches the one on the label/sticker/case of your device and tell me the device-tree path leading to it. Note that you do not have to post your MAC addresses here, but just a statement about which match, e.g. like
```
/proc/device-tree/ahb/eth@1a000000/mac-address
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment
/proc/device-tree/ahb/eth@1a000000/mtd-mac-address
/proc/device-tree/ahb/eth@19000000/mac-address <== this one
/proc/device-tree/ahb/eth@19000000/mtd-mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mac-address
/proc/device-tree/ahb/apb/wmac@18100000/mtd-mac-address
```

There may be one, two or no paths giving the correct address. It may also happen that the `find` command returns nothing. Thanks.

adrianschmutzler commented:
Unfortunately, my description is slightly incorrect. The MAC addresses are stored binary, so you can't read with `cat`, but have to use the following code to read from the paths retrieved by `find`:
```
. /lib/functions/system.sh
get_mac_binary "$yourpath" 0
```
The include is obviously only required once. Sorry.

marcin1j commented:
I have (and this is not going to change either way) an occasional access to the device, because it's installed in a remote location.

Device tree shows the following MAC addresses:

|  Path | MAC |
|---------|---------|
|  `/proc/device-tree/ahb/wmac@18100000/mac-address` | `AA:BB:CC:DD:EE:d6` |
|  `/proc/device-tree/ahb/wmac@18100000/mtd-mac-address` | `00:00:00:07:00:01` |
|  `/proc/device-tree/ahb/eth@1a000000/mac-address` | `AA:BB:CC:DD:EE:d5` |
|   `/proc/device-tree/ahb/eth@1a000000/mtd-mac-address-increment` | `ff:ff:ff:ff:  :  ` |
|   `/proc/device-tree/ahb/eth@1a000000/mtd-mac-address` | `00:00:00:07:00:01` |
|  ` /proc/device-tree/ahb/eth@19000000/mac-address` | `AA:BB:CC:DD:EE:d6` |
|   `/proc/device-tree/ahb/eth@19000000/mtd-mac-address` | `00:00:00:07:00:01` |


I'll read the sticker and let you know which one of them is factory MAC address in a day or two.

CodeFetch commented:
@adrianschmutzler @marcin1j TP-Link devices have only one MAC address stored at 0x1fc00 on the flash chip. This seems to always be the MAC address on the label . The other MAC addresses are in most cases incremented or decremented by one or two.

```
mtd-mac-address = <&uboot 0x1fc00>;
mtd-mac-address-increment = <(-1)>;
```

For OpenWrt pull-requests for new device support thus it always needs to be tested what the MAC address of each network device is in stock firmware. E.g. the WAN MAC address in OpenWrt should always match the WAN MAC address in stock firmware. Unfortunately this is not always the case as there is no proper device support checklist and devs don't remind people to check this...

In ar71xx nbd checked the MAC addresses for the WR710N v1. I trust him that he respected this.

Thus:
eth0 = label MAC address
eth1 = label MAC address - 1
wmac = label MAC address

adrianschmutzler commented:
@CodeFetch I would also expect eth0 and wmac to match the label here, and already have implemented it that way in my PR based on the general feedback about those devices. However, having seen a lot of random choices dealing with this subject, I'm still glad for any real-life evidence I get.

marcin1j commented:
MAC address on the sticker is `AA:BB:CC:DD:EE:d6`. This matches:

- `/proc/device-tree/ahb/wmac@18100000/mac-address`
- `/proc/device-tree/ahb/eth@19000000/mac-address`

 The corresponding network interface is `eth1`, which is LAN (MAC address for `eth0`, which is WAN: `AA:BB:CC:DD:EE:d5`).

adrianschmutzler commented:
Thanks, so this device will have `label-mac-device = &eth0;`.

CodeFetch commented:
@adrianschmutzler Can you please explain this? marcin1j said it's eth1. Is this router using physwap?

adrianschmutzler commented:
For label-mac-device, I need the name of the node in DTS. That's just:
eth0: /proc/device-tree/ahb/eth@19000000/mac-address
eth1: /proc/device-tree/ahb/eth@1a000000/mac-address

Based on the data from marcin1j, the address is stored in `/proc/device-tree/ahb/eth@19000000/mac-address`, so that will be &eth0.

eth0/eth1 during runtime is a different story, (I think) names there depend on initialization order. There has been a patch/PR to fix this for many devices (https://github.com/openwrt/openwrt/pull/2066), but still e.g. GLinet AR150 has inverse eth0/eth1 order and some other devices, too (I also think C58/C59).
However, if MAC addresses are available in DTS, in do not care about that, as it won't touch my goal to get the label-mac-address.

But this situation is the reason why implementing label-mac-device is not as easy as just copying data from Gluon's primary MAC address and why I insist on users retrieving device-tree information for so many devices.

marcin1j commented:
Both GLinet and TL-WR 710 have swapped network inteface numbers. PR #2066 aimed at fixing this for a couple of devices and generally did the job but has failed for above-mentioned two models.

Taking these two devices into account, either `label-mac-device = &eth1;` or `label-mac-device = &wmac;` would be fine. Once again, `eth0` is WAN and corresponding MAC address `AA:BB:CC:DD:EE:d5` is not what you want.

adrianschmutzler commented:
@marcin1j No, `/proc/device-tree/ahb/eth@19000000/mac-address` gives `AA:BB:CC:DD:EE:d6` and this one is &eth0 in DTS (no matter what it is on the running device):
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/ath79.dtsi#L43
(and, of course, `label-mac-device = &wmac;` is also possible)
